### PR TITLE
fix: ran sb upgrade and fixed peer deps

### PR DESCRIPTION
**Pull request by dannyhw**
I ran sb upgrade to update to latest 6.4 rc and I also made the peer deps more reasonable.

the eslint plugin was suggested by the upgrade command so I applied it also.

I also ran `npx yarn-audit-fix` since I noticed the dependabot warnings.

<details>
  <summary>📦 Published PR as canary version: <code>0.0.7-canary.11.34446b7.0</code></summary>
  <br />
  
  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @storybook/addon-react-native-web@0.0.7-canary.11.34446b7.0
  # or 
  yarn add @storybook/addon-react-native-web@0.0.7-canary.11.34446b7.0
  ```
</details>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -91,11 +93,11 @@
...
-    "@storybook/addons": "6.4.0-beta.11",
-    "@storybook/api": "6.4.0-beta.11",
-    "@storybook/components": "6.4.0-beta.11",
-    "@storybook/core-events": "6.4.0-beta.11",
-    "@storybook/theming": "6.4.0-beta.11",
+    "@storybook/addons": "^6 || ^6.4.0-beta",
+    "@storybook/api": "^6 || ^6.4.0-beta",
+    "@storybook/components": "^6 || ^6.4.0-beta",
+    "@storybook/core-events": "^6 || ^6.4.0-beta",
+    "@storybook/theming": "^6 || ^6.4.0-beta",
```
Comment (shilman) [inline package.json:100]: `^6.4.0-beta` doesn't match anything unfortunately

**Review by shilman — CHANGES_REQUESTED** [review]
Minor issue

File: package.json (hunk 1 above)
Comment (dannyhw) [reply package.json]: ok I just updated it to `^6 || 6.4.0-rc.1` what do you think?

**Review by shilman — APPROVED** [review]
LGTM!

**Pull request merged**